src/main/java/thewaypointers/trafficsimulator/TrafficSimulatorManager.java

```diff
@@ -1,31 +1,60 @@
 package thewaypointers.trafficsimulator;
+import javax.swing.JButton;
 
+import thewaypointers.trafficsimulator.common.IStateProvider;
 import thewaypointers.trafficsimulator.common.WorldStateDTO;
 import thewaypointers.trafficsimulator.gui.MainFrame;
-import thewaypointers.trafficsimulator.simulation.Simulation;
 
-public class TrafficSimulatorManager {
+public class TrafficSimulatorManager extends Thread {
 
-    final static long TIME_STEP = 100;
-    final static long SIMULATION_TIME_STEP = 300;
-    static WorldStateDTO worldState;
-    static MainFrame mainFrame;
+    public static final float VEHICLE_MOVEMENT_SPEED = 2;
+    public static final float STATES_PER_SECOND = 1;
+    private static MainFrame mainFrame;
+    private static IStateProvider simulation;
+    private WorldStateDTO worldState;
+    private boolean stop = true;
+    private boolean isSleep = true;
+    private JButton start_pauseButton;
 
-    public static void main(String[] args) {
 
-        Simulation simulation = new Simulation();
-        mainFrame = new MainFrame();
+    public static void setSimulation(IStateProvider simulation) {
+        TrafficSimulatorManager.simulation = simulation;
+    }
 
-        while(true){
-            try{
-                Thread.sleep(TIME_STEP);
-                worldState = simulation.getNextSimulationStep(SIMULATION_TIME_STEP);
-                MainFrame.mapContainerPanel.mapPanel.NewStateReceived(worldState);
-            }
-            catch(InterruptedException ex){
-                System.out.println(ex.getMessage());
+    public static void setMainFrame(MainFrame mainFrame) {
+        TrafficSimulatorManager.mainFrame = mainFrame;
+    }
+
+    public TrafficSimulatorManager(JButton start_pause){
+        start_pauseButton = start_pause;
+    }
+
+    public void changeState() {
+        isSleep = !isSleep;
+        start_pauseButton.setText("Start");
+    }
+
+    public void stopState() {
+        stop = !stop;
+    }
+
+    public void run() {
+        while (true) {
+            if (stop) {
```

Comment:
What's the difference between `stop` and `isSleep`? It looks like they work exactly the same way.